sapi/cli/php_cli_server.c

```diff
@@ -348,18 +348,34 @@ static void append_http_status_line(smart_str *buffer, int protocol_version, int
 	smart_str_appendl_ex(buffer, "\r\n", 2, persistent);
 } /* }}} */
 
-static void append_essential_headers(smart_str* buffer, php_cli_server_client *client, int persistent) /* {{{ */
+static void append_essential_headers(smart_str* buffer, php_cli_server_client *client, int persistent, sapi_headers_struct *sapi_headers) /* {{{ */
 {
 	char *val;
 	struct timeval tv = {0};
+	sapi_header_struct *h;
+	zend_llist_position pos;
+	bool append_date_header = 1;
+
+	if (sapi_headers != NULL) {
+		h = (sapi_header_struct*)zend_llist_get_first_ex(&sapi_headers->headers, &pos);
+		while (h) {
+			if (h->header_len) {
```

Comment:
This can be improved further by changing this to: `if (h->header_len > strlen("Date:")) {` such that the more expensive string comparison is only done if it's useful. This is similar to how `cgi_main.c` checks headers.

Reply:
I used `strlen("Date:")-1`, like in `cgi_main.c` to avoid missing an empty Date header